src/repr/src/relation.rs

```diff
@@ -355,6 +359,65 @@ impl proptest::arbitrary::Arbitrary for ColumnName {
     }
 }
 
+/// Stable index of a column in a [`RelationDesc`].
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Hash, MzReflect,
+)]
+pub struct ColumnIndex(usize);
+
+static_assertions::assert_not_impl_all!(ColumnIndex: Arbitrary);
+
+/// The version a given column was added at.
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize, Hash, MzReflect,
+)]
+pub struct RelationVersion(u64);
+
+impl RelationVersion {
+    pub fn root() -> Self {
+        RelationVersion(0)
+    }
+
+    /// Returns an instance of [`RelationVersion`] which is "one" higher than `self`.
+    pub fn bump(&self) -> Self {
+        let next_version = self
+            .0
+            .checked_add(1)
+            .expect("added more than u64::MAX columns?");
+        RelationVersion(next_version)
+    }
+
+    /// Returns the inner value.
+    ///
+    /// TODO(parkmycar): Remove this [`RelationVersion`] should be opaque.
```

Comment:
Where are we using this now? I didn't see any callers.

Reply:
It's used in a stacked PR, removed it for now though!